packages/lib/src/components/UPI/components/UPIComponent/UPIComponent.tsx

```diff
@@ -69,23 +77,25 @@ export default function UPIComponent({ defaultMode, onChange, onUpdateMode, payB
 
             setSelectedApp(app);
             setIsValid(true);
+            setStatus('success');
         },
         [selectedApp]
     );
 
-    const handleVpaInputChange = useCallback(({ data: { virtualPaymentAddress }, errors, valid, isValid }: OnChangeProps) => {
-        setVpa(virtualPaymentAddress);
-        setErrors(errors);
-        setValid(valid);
-        setIsValid(isValid);
-    }, []);
+    const validateIntentApp = useCallback(() => {
+        if (!selectedApp) {
+            setStatus('error');
+            setIsValid(false);
+        } else {
+            setStatus('success');
+            setIsValid(true);
+        }
+    }, [selectedApp]);
 
     useEffect(() => {
         if (mode !== 'qrCode') {
             onChange({
-                data: { ...(vpa && { virtualPaymentAddress: vpa }), ...(selectedApp && { app: selectedApp }) },
-                errors,
-                valid,
+                data: { ...(selectedApp && { app: selectedApp }) },
```

Comment:
Don't we need to pass the `virtualPaymentAddress`  anymore in case it is available?